robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowAssetManager.java

```diff
@@ -103,8 +103,13 @@ private void convertAndFill(AttributeResource attribute, TypedValue outValue, St
         throw new Resources.NotFoundException("unknown resource " + resName);
       }
       outValue.type = TypedValue.TYPE_REFERENCE;
-      outValue.resourceId = resourceId;
+      if (!resolveRefs) {
+          // Just return the resourceId if resolveRefs is false.
+          outValue.data = resourceId;
+          return;
+      }
 
+      outValue.resourceId = resourceId;
```

Comment:
Move this above the early return; the resourceId field should be filled regardless of whether references are resolved or not.